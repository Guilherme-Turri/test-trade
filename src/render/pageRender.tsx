import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};