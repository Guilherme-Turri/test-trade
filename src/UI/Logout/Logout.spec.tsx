import { screen } from '@testing-library/react'
import { Logout } from '.'
import { componentRender } from '../../render/componentRender'

describe('</Logout>', () => {
  it('should render on screen', () => {
    const { container } = componentRender(<Logout />)
    expect(screen.getByTestId('Logout')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
})