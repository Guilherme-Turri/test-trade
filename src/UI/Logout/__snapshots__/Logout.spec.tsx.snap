// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Logout> should render on screen 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  color: #00c555;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="c0"
    data-testid="Logout"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M336 376V272H191a16 16 0 0 1 0-32h145V136a56.06 56.06 0 0 0-56-56H88a56.06 56.06 0 0 0-56 56v240a56.06 56.06 0 0 0 56 56h192a56.06 56.06 0 0 0 56-56zm89.37-104-52.68 52.69a16 16 0 0 0 22.62 22.62l80-80a16 16 0 0 0 0-22.62l-80-80a16 16 0 0 0-22.62 22.62L425.37 240H336v32z"
      />
    </svg>
  </div>
</div>
`;
